express/parsers/apps/aiida/formats/zip.py

```diff
@@ -0,0 +1,171 @@
+        return self._metadata
+
+    @property
+    def data(self):
+        if self._data is None:
+            with ZipFile(self.zip_file_path) as source:
+                self._data = json.loads(source.read('data.json'))
+        return self._data
+
+    def _check_supported_versions(self):
+        """
+        Check whether the versions of the exported archive file are supported.
+
+        Args:
+            metadata (dict): deserialized metadata.json object
+
+        Raises:
+            RuntimeError
+        """
+        try:
+            aiida_version = self.metadata['aiida_version']
+            export_version = self.metadata['export_version']
+        except KeyError as error:
+            raise BadAiidaArchiveZipFile(self.zip_file_path, error)
+
+        if self.metadata['aiida_version'] != SUPPORTED_AIIDA_VERSION:
+            raise AiidaArchiveFileParseError(
+                path=self.zip_file_path,
+                reason=f"aiida_version ({metadata['aiida_version']}) not supported. "
+                f"Supported: {SUPPORTED_AIIDA_VERSION}")
+
+        if self.metadata['export_version'] != SUPPORTED_AIIDA_ARCHIVE_VERSION:
+            raise AiidaArchiveFileParseError(
+                path=self.zip_file_path,
+                rason=f"archive export_version ({metadata['export_version']}) not supported. "
+                f"Supported: {SUPPORTED_AIIDA_ARCHIVE_VERSION}")
+
+    def structures(self):
+        """
+        Extract all structures from compatible AiiDA archive zip files.
+
+        Returns:
+            list
+        """
+
+        return list(self._gather_structures(self.data))
+
+    @classmethod
+    def _gather_structures(cls, data):
+        """
+        Yield parsed structure objects from nodes within AiiDA archive.
+
+        Args:
+            data (dict): deserialized data.json object
+        Yields:
+            dict
+        """
+
+        nodes = data['export_data']['Node']
+        structure_nodes = {pk: node for (pk, node) in nodes.items()
+                           if node['node_type'] == 'data.structure.StructureData.'}
+
+        for pk in structure_nodes:
+            export_data = data['export_data']['Node'][pk]
+            node_attributes = data['node_attributes'][pk]
+            yield cls._parse_structure_node_attributes(export_data, node_attributes)
+
+    @staticmethod
+    def _parse_structure_node_attributes(export_data, attributes):
+        """
+        Parse an AiiDA StructureData node.
+        """
+
+        # Prepare mapping
+        assert export_data['node_type'] == 'data.structure.StructureData.'
+
+        sites = list(enumerate(attributes['sites']))
+        kinds = {kind['name']: kind for kind in attributes['kinds']}
+        assert all(len(kind['symbols']) == 1 for kind in kinds.values())
+
+        instance = json.loads(TEMPLATES_MATERIALS_PATH.read_text())
+        instance.update({
+            '_id': export_data['uuid'],
+            'created_at': export_data['ctime'],
```

Comment:
Also - when we upload the job information to our database online, the ID is likely to be reset, especially if it's a duplicate. We could add a tag to store the original AiiDA ids there. With MP materials we retain them as `mp-137`, for example. So `tags: ["aiida-<UUID>"]` could be a good start.

Reply:
I applied that change in https://github.com/Exabyte-io/express/pull/56/commits/3c01bbb58439a40222011b24349f694c6f539d66 .